src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -2181,7 +2196,7 @@ private String initializeDeviceId() {
 
         // see if device id already stored in db
         String deviceId = dbHelper.getValue(DEVICE_ID_KEY);
-        if (!(Utils.isEmptyString(deviceId) || invalidIds.contains(deviceId))) {
+        if (!(Utils.isEmptyString(deviceId) || invalidIds.contains(deviceId) || deviceId.endsWith("S"))) {
```

Comment:
This is based on https://developer.android.com/training/articles/app-set-id#dont-rely-on-cached-value , then we just fetch each time.